src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -25,252 +26,1010 @@
+  assert((last_idx >= 0) && (last_idx < _num_bits), "precondition");
+  assert(last_idx - count >= -1, "precondition");
+  size_t array_idx = last_idx >> LogBitsPerWord;
+  uintx bit_number = last_idx & right_n_bits(LogBitsPerWord);
+  uintx element_bits = _bitmap[array_idx];
+  uintx bits_to_examine = bit_number + 1;
+  element_bits <<= (BitsPerWord - bits_to_examine);
+  uintx complement = ~element_bits;
+  uintx leading_ones;
+  if (complement) {
+    leading_ones = count_leading_zeros<uintx>(complement);
+  } else {
+    leading_ones = bits_to_examine;
+  }
+  if (leading_ones >= (uintx) count) {
+    return true;
+  } else if (leading_ones == bits_to_examine) {
+    // Tail recursion
+    return is_backward_consecutive_ones(last_idx - leading_ones, count - leading_ones);
+  } else {
+    return false;
+  }
+}
+
+ssize_t ShenandoahSimpleBitMap::find_next_consecutive_bits(size_t num_bits, ssize_t start_idx, ssize_t boundary_idx) const {
+  assert((start_idx >= 0) && (start_idx < _num_bits), "precondition");
+
+  // Stop looking if there are not num_bits remaining in probe space.
+  ssize_t start_boundary = boundary_idx - num_bits;
+  if (start_idx > start_boundary) {
+    return boundary_idx;
+  }
+  uintx array_idx = start_idx >> LogBitsPerWord;
+  uintx bit_number = start_idx & right_n_bits(LogBitsPerWord);
+  uintx element_bits = _bitmap[array_idx];
+  if (bit_number > 0) {
+    uintx mask_out = right_n_bits(bit_number);
+    element_bits &= ~mask_out;
+  }
+  while (true) {
+    if (!element_bits) {
+      // move to the next element
+      start_idx += BitsPerWord - bit_number;
+      if (start_idx > start_boundary) {
+        // No match found.
+        return boundary_idx;
+      }
+      array_idx++;
+      bit_number = 0;
+      element_bits = _bitmap[array_idx];
+    } else if (is_forward_consecutive_ones(start_idx, num_bits)) {
+      return start_idx;
+    } else {
+      // There is at least one non-zero bit within the masked element_bits.  Find it.
+      uintx next_set_bit = count_trailing_zeros<uintx>(element_bits);
+      uintx next_start_candidate_1 = (array_idx << LogBitsPerWord) + next_set_bit;
+
+      // There is at least one zero bit in this span.  Align the next probe at the start of trailing ones for probed span.
+      size_t trailing_ones = count_trailing_ones(start_idx + num_bits - 1);
+      uintx next_start_candidate_2 = start_idx + num_bits - trailing_ones;
+
+      start_idx = MAX2(next_start_candidate_1, next_start_candidate_2);
+      if (start_idx > start_boundary) {
+        // No match found.
+        return boundary_idx;
+      }
+      array_idx = start_idx >> LogBitsPerWord;
+      element_bits = _bitmap[array_idx];
+      bit_number = start_idx & right_n_bits(LogBitsPerWord);
+      if (bit_number > 0) {
+        size_t mask_out = right_n_bits(bit_number);
+        element_bits &= ~mask_out;
+      }
+    }
+  }
+}
+
+ssize_t ShenandoahSimpleBitMap::find_prev_consecutive_bits(
```

Comment:
It looks weird to me to have an opening ( and then all the args on the next line. I think in HotSpot, it's usually done like:
```
ssize_t ShenandoahSimpleBitMap::find_prev_consecutive_bits(const size_t num_bits,
                                                           ssize_t last_idx,
                                                           const ssize_t boundary_idx) const {
```

Reply:
Fixed